lib/puppet/type/cisco_vxlan_vtep_vni.rb

```diff
@@ -0,0 +1,185 @@
+# Manages a Cisco Virtual Tunnel Endpoint (VTEP) to Virtual Network
+# Identifier (VNI) binding.
+#
+# January 2016
+#
+# Copyright (c) 2013-2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'ipaddr'
+begin
+  require 'puppet_x/cisco/cmnutils'
+rescue LoadError # seen on master, not on agent
+  # See longstanding Puppet issues #4248, #7316, #14073, #14149, etc. Ugh.
+  require File.expand_path(File.join(File.dirname(__FILE__), '..', '..',
+                                     'puppet_x', 'cisco', 'cmnutils.rb'))
+end
+
+Puppet::Type.newtype(:cisco_vxlan_vtep_vni) do
+  @doc = "Manages a Cisco Virtual Tunnel Endpoint (VTEP) to Virtual Network
+  Identifier (VNI) mapping.
+
+  cisco_vxlan_vtep_vni {'<title>':
+    ..attributes..
+  }
+
+  <title> is the title of vxlan_vtep_vni resource.
+
+  Example:
+    cisco_vxlan_vtep_vni {'nve1 10000':
+      ensure              => present,
+      assoc_vrf           => false,
+      ingress_replication => 'static',
+      peer_list           => ['1.1.1.1', '2.2.2.2', '3.3.3.3'],
+      suppress-arp        => false,
+    }
+
+    cisco_vxlan_vtep_vni {'nve1 20000':
+      ensure              => present,
+      assoc_vrf           => false,
+      multicast_group     => '224.1.1.1'
+      suppress-arp        => true,
+    }
+"
+
+  ###################
+  # Resource Naming #
+  ###################
+
+  # Parse out the title to fill in the attributes in these
+  # patterns. These attributes can be overwritten later.
+  def self.title_patterns
+    identity = ->(x) { x }
+    patterns = []
+
+    # Below pattern matches both parts of the full composite name.
+    patterns << [
+      /^(\S+)\s+(\d+)$/,
+      [
+        [:interface, identity],
+        [:vni, identity],
+      ],
+    ]
+    patterns
+  end
+
+  newparam(:interface, namevar: :true) do
+    desc 'Name of the nve interface on the network element.
+          Valid values are string.'
+
+    validate do |value|
+      fail('Interface name must be a string') unless value.is_a? String
+    end
+
+    munge(&:downcase)
+  end
+
+  newparam(:vni, namevar: true) do
+    desc 'ID of the Virtual Network Identifier. Valid values are integer.'
```

Comment:
This should be able to handle an integer as a `fixnum` or `string`.  Just munch anything that comes in to be an integer.

Example:

https://github.com/cisco/cisco-network-puppet-module/blob/develop/lib/puppet/type/cisco_ospf_vrf.rb#L114